File: jcli/src/com/maxima/bbgum/Action.java
package com.maxima.bbgum;

public class Action {

    private byte id;
    private byte transnum;
    private byte[] data;

    public Action() {

    }

    public Action(final byte[] data) throws FrameError {

        if (data.length > Frame.FRAME_BODY_MAX_LENGTH) {
            String msg = "Action can not be bigger than " + Frame.FRAME_BODY_MAX_LENGTH + " " + "bytes";
            throw new FrameError(msg);
        }

        this.setId(data[0]);
        this.setTransNum(data[1]);

        byte dataForAction[] = new byte[data.length - Frame.ACTION_FLOW_INFO_SEGMENT_LENGTH];
        System.arraycopy(data,
            Frame.ACTION_FLOW_INFO_SEGMENT_LENGTH,
            dataForAction, 0, dataForAction.length);
        this.setData(dataForAction);
    }

    public byte getId() {
        return id;
    }

    public void setId(byte id) {
        this.id = id;
    }

    public byte getTransNum() {
        return transnum;
    }

    public void setTransNum(byte transaction) {
        this.transnum = transaction;
    }

    public byte[] getData() {
        return data;
    }

    public void setData(byte[] data) {
        this.data = data;
    }
}

File: bbgum/action.py
from bbgum.frame import FrameError, Frame

class Action(object):

    self.archetype = b'\x00'
    self.transnum = b'\x00'
    self.buffer = bytearray()

    def __init__(self, data = None):
        byteszero = lambda n: bytearray([0] * n)
        if data:
            length = len(data)
            if (length) > Frame.FRAME_BODY_MAX_LENGTH):
                msg = "Action can not be bigger than " + str(Frame.FRAME_BODY_MAX_LENGTH) + " " + "bytes";
                raise FrameError(msg)
            self.archetype = (data[0])
            self.transnum = (data[1])
            self.buffer = data[Frame.ACTION_FLOW_INFO_SEGMENT_LENGTH:-1]
